### ISSUE TITLE
Port kgraft-analysis-tool to klp-build

### ISSUE THREAD

**Issue by fgyanz** [SUSE/klp-build#47]
We need a way to automatically identify when and where a given function symbol is being inlined. Due to the huge amount of kernels that we currently support, it's unrealistic and a waste of time to manually check such things. Porting, if possible, [kgraft-analysis-tool](https://github.com/marxin/kgraft-analysis-tool/tree/master) to klp-build would immensely simplify and ease the workload of the livepatch developers.

**Comment by marcosps** [comment]
I believe that we need to merge this tool soon, so we can risk this item on our TODO list.

**Comment by marcosps** [comment]
We can start working on it after #62 is merged

**Comment by vmezzela** [comment]
Currently, the integration of `kgraft-analysis-tool` is not really needed in `klp-build`, since `klp-ccp` (or rather its policy) is already able to perform [ipa-clones analysis](https://github.com/SUSE/klp-ccp/blob/master/examples/policies/suse/ipa_clones.py). There were minor issues why this was not working properly that are currently [being worked](https://github.com/SUSE/klp-ccp/pull/9) .

Once those issues are solved we should already see `klp-ccp` extracting the callers of the inlined functions with no further changes to `klp-build`, although the set of functions to livepatch would not be reflected yet to the relative `patched_funcs.csv`.

The idea would be to let the suse `klp-ccp` policy to write the `patched_funcs.csv` instead of `klp-build`. This action item would require some changes to `klp-build` as well.

Anyway, before proceeding with this, it's necessary to integrate some further analysis (on the DWARF) that would be complementary to the ipa-clones ones.

**Comment by marcosps** [comment]
@vmezzela for now I believe that we can close this issue, since the work being done by kgraft-analysis is being done by klp-ccp, and now we can check whenever a symbols is inlined and where to it is inlined, right?

**Comment by vmezzela** [comment]
You are right, `klp-ccp` is now handling the inlining problem (only on `x86_64`) and this means that most of the time we won't need to check them manually. All the analysis is being done on the original functions to livepatch during the extraction phase.

This means that the whole analysis is automated in this step, so we can't manually check the inlining of a specific function. Before closing this issue I wonder though if there is still any use case for which we would need to analyze the inlining manually.

What do you think?

**Comment by marcosps** [comment]
I don't think so. If we find a function that is inlined on the extractiong part, we should go back and run setup again with the `correct` function.
